### ISSUE TITLE
[rllib] Synchronizing policy state with outer model

### ISSUE THREAD

**Issue by nrhine1** [ray-project/ray#10326]
I have an auxiliary model, `outer_M` (which is an instance of a `nn.Module`-derived class), that I am updating as part of a `custom_loss` function. I have added a new attribute to the policy, `self.M = outer_M`. `outer_M` is instantiated before any Policy or Trainer is instantiated, specifically in my `main()` function, before `ray.tune` is called. I have noticed that `id(outer_M)`, before I initialize the policy, is different than `id(self.M)`, inside the `make_model_and_action_dist` custom model creation function for my policy. This means that updates to `self.M` (which occur as part of optimization of the `custom_loss` function) are not reflected in the original auxiliary model, `outer_M`.

I am using `ray.init(local=True)` to make this issue easier to debug.

Why is the id different (is it because of some underlying wrapping or remoting going on in `ray.tune` or `Trainer.train()`?) What is a good way to ensure that the updates to `self.M`, which occur as part of optimization of the `custom_loss` function, are properly synchronized to the `outer_M`?

I have also tried making `outer_M`, and therefore `self.M`, a `ray.Actor`, but this didn't seem to work -- the optimizer instantiated via `torch.optim.Adam(ray.get(self.M.parameters_list.remote()), lr=lr)` ends yielding `None` gradients for the auxiliary loss function, possibly because these parameters are copies(?) of the underlying parameters? Or possibly because the auxiliary loss function is also computed via `self.M.compute_loss.remote(...)`? 

**Edit**: I found a way to do this -- by overriding the `set_weights` and `get_weights` member functions with a mixin that syncs a locally-instantiated model (`self.M`) to a remote actor `self.M_actor`, I am able to sync out the learned model so that it can be used elsewhere
